src/multiprocess/multiprocess_memory_limit.c

```
@@ -118,10 +118,11 @@ size_t get_limit_from_env(const char* env_name) {
     size_t res = strtoul(env_limit, &digit_end, 0);
     size_t scaled_res = res * scalar;
     if (scaled_res == 0) {
-        if (env_name[12]=='S'){
+        size_t name_len = strlen(env_name);
+        if (name_len > 12 && env_name[12]=='S'){
             LOG_INFO("device core util limit set to 0, which means no limit: %s=%s",
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)